src/check-couch-up.js

```diff
@@ -0,0 +1,51 @@
+const utils = require('./utils');
+const NUM_RETRY = 100;
+const TIMEOUT_RETRY = 1000; // 1 second
+
+const isCouchUp = async () => {
+  const url = utils.getUrl('/');
+  try {
+    await utils.request({ url });
+    return true;
+  } catch (err) {
+    return false;
+  }
+};
+
+const isClusterReady = async (expectedNodes) => {
+  const nodes = await utils.getNodes();
+  return nodes && nodes.length === expectedNodes;
+};
+
+const repeatRetry = async (promiseFn) => {
+  let fulfilled;
+  let retry = NUM_RETRY;
+  do {
+    fulfilled = await promiseFn();
+    if (!fulfilled) {
+      retry--;
+      await new Promise(r => setTimeout(r, TIMEOUT_RETRY));
+    }
+  } while (retry > 0 && !fulfilled);
+
+  return fulfilled;
```

Comment:
Short circuit to drop the variable...

```suggestion
  let retry = NUM_RETRY;
  do {
    if (await promiseFn()) {
      return true;
    }
    retry--;
    await new Promise(r => setTimeout(r, TIMEOUT_RETRY));
  } while (retry > 0);
  return false;
```